src/bitmap/store/array_store/mod.rs

```diff
@@ -205,6 +205,23 @@ impl ArrayStore {
     pub fn as_slice(&self) -> &[u16] {
         &self.vec
     }
+
+    /// Retains only the elements specified by the predicate.
+    pub fn retain(&mut self, mut f: impl FnMut(u16) -> bool) {
```

Comment:
Should this be unsafe? If the supplied `f` doesn't uphold the invariant 
> // SAFETY: pos is always at most i because `f(val) as usize` is at most 1.

I suspect this will lead to undefined behavior.

Reply:
`f(val)` is `bool`

Reply:
Ooops didn't look at the return type in the signature.